### DISCUSSION TITLE
How do I access the headers I set in middleware?

### DISCUSSION THREAD

**Discussion by WillPwn4Food** [vercel/next.js#39300] [Help]
So with middleware I can set custom headers by doing something like: `response.headers.set('x-utm-date', utmDate);`, that seems easy enough, similar to how I set the cookies, which works.

Now when I load my static page, I'd love to see those custom headers I just sent over so I can take action on them, but I have no idea how to access them or if it's even possible. I can't find any documentation on this. I assume I would have to do it in `useEffect` or is it somehow passed in as a `prop` that I don't know about?

Please let me know if you have any ideas. If I can't access it, what's the point of setting the headers?
Thanks,
- Ivan

**Comment by pjaws** [comment]
Not positive, but it sounds like the issue is that you're trying to use `getStaticProps` instead of `getServerSideProps`. The former is going to be called at build time and serve static HTML/JS. It shouldn't be used in situations where you need to read some variable set at runtime on every request.
